src/lib/reasoners/enum_rel.ml

```diff
@@ -44,11 +46,26 @@ module HSS = Set.Make (struct type t=Hs.t let compare = Hs.compare end)
 
 module LR = Uf.LX
 
-type t = { mx : (HSS.t * Ex.t) MX.t; classes : Expr.Set.t list;
-           size_splits : Numbers.Q.t }
+type t = {
+  mx : (HSS.t * Ex.t) MX.t;
+  (* Map of uninterpreted enum semantic values to domains of their possible
+     values. The explanation justifies that the value of the semantic value
+     lies in the domain. *)
 
-let empty classes = { mx = MX.empty; classes = classes;
-                      size_splits = Numbers.Q.one }
+  classes : Expr.Set.t list;
+  (* State of the union-find represented by its equivalence classes.
+     The field is updated while assuming literals of the theory. *)
```

Comment:
It's actually the same design in most theories. We have to save the state of the all union-find after assuming each literal in order to return the appropriate state if we perform a query in this theory and we got an inconsistency.

Reply:
Actually, it's the case in all theories, which means we can bubble this feature in `Relation`  but I'm not sure it will be clearer.

Reply:
Ah right, this for caching is the result of `cl_extract` that is used only for a debug message. I don't think we can lift this to `Relation` (it is needed to raise `Ex.Inconsistent` and the relations do raise `Ex.Inconsistent`). We should make it clear in the doc that this is a cache of `Uf.cl_extract` (if the `--bottom-classes` flag is not passed to Alt-Ergo, `Uf.cl_extract` is always `[]`).

Reply:
That's why I said `I'm not sure it will be clearer` as we would have to create a new intermediate exception :/ 
I agree that the best solution is to write a clear comment.